src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/subjectaccessrequests/SubjectAccessRequestService.kt

```diff
@@ -0,0 +1,167 @@
+    val bookingExtensions = cas3SubjectAccessRequestRepository.bookingExtensions(crn, nomsNumber, startDate, endDate, ServiceName.temporaryAccommodation)
+    val cancellations = cas3SubjectAccessRequestRepository.cancellations(crn, nomsNumber, startDate, endDate, ServiceName.temporaryAccommodation)
+    val domainEvents = cas3SubjectAccessRequestRepository.domainEvents(crn, nomsNumber, startDate, endDate, "CAS3")
+    val domainEventsMetaData = cas3SubjectAccessRequestRepository.domainEventMetadata(crn, nomsNumber, startDate, endDate, "CAS3")
+
+    val result = """
+      {
+        "Applications": $temporaryAccommodationApplications,
+        "Assessments": $temporaryAccommodationAssessments,
+        "AssessmentReferralHistoryNotes": $assessmentReferralHistoryNotes,
+        "Bookings": $bookings,
+        "BookingExtensions": $bookingExtensions,
+        "Cancellations": $cancellations,
+        "DomainEvents": $domainEvents,
+        "DomainEventsMetadata": $domainEventsMetaData
+      }
+    """.trimIndent()
+
+    if (log.isDebugEnabled) {
+      val prettyPrintJson = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(
+        objectMapper.readValue(result, Any::class.java),
+      )
+      log.debug("CAS3 SAR result is $prettyPrintJson")
+    }
+    return result
+  }
+
+  fun getCAS2Result(crn: String?, nomsNumber: String?, startDate: LocalDateTime?, endDate: LocalDateTime?): String {
+    val applications = cas2SubjectAccessRequestRepository.getApplicationsJson(crn, nomsNumber, startDate, endDate)
+    val applicationNotes = cas2SubjectAccessRequestRepository.getApplicationNotes(crn, nomsNumber, startDate, endDate)
+    val statusUpdates = cas2SubjectAccessRequestRepository.getStatusUpdates(crn, nomsNumber, startDate, endDate)
+    val statusUpdateDetails = cas2SubjectAccessRequestRepository.getStatusUpdateDetails(crn, nomsNumber, startDate, endDate)
+    val assessments = cas2SubjectAccessRequestRepository.getAssessments(crn, nomsNumber, startDate, endDate)
+    val domainEvents = cas2SubjectAccessRequestRepository.domainEvents(crn, nomsNumber, startDate, endDate, "CAS2")
+    val domainEventsMetaData = cas2SubjectAccessRequestRepository.domainEventMetadata(crn, nomsNumber, startDate, endDate, "CAS2")
+
+    val result = """
+     {
+        "Applications": $applications,
+        "ApplicationNotes": $applicationNotes,
+        "Assessments": $assessments,
+        "StatusUpdates": $statusUpdates,
+        "StatusUpdateDetails": $statusUpdateDetails,
+        "DomainEvents": $domainEvents,
+        "DomainEventsMetadata": $domainEventsMetaData
+     }
+    """.trimIndent()
+
+    if (log.isDebugEnabled) {
+      val prettyPrintJson = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(
+        objectMapper.readValue(result, Any::class.java),
+      )
+      log.debug("CAS2 SAR result is $prettyPrintJson")
+    }
```

Comment:
we repeat this code a few times. Could we maybe introduce a private function or extension function to pretty print a given string (i.e. result)

Reply:
nice! let me take a look.